Docker.DotNet/DockerClient.cs

```diff
@@ -74,31 +80,50 @@ internal Task<Stream> MakeRequestForStreamAsync(IEnumerable<ApiResponseErrorHand
 
         internal async Task<DockerApiResponse> MakeRequestAsync(IEnumerable<ApiResponseErrorHandlingDelegate> errorHandlers, HttpMethod method, string path, IQueryString queryString, IRequestContent data, TimeSpan? timeout, CancellationToken cancellationToken)
         {
-            HttpResponseMessage response = await this.MakeRequestInnerAsync(null, HttpCompletionOption.ResponseContentRead, method, path, queryString, null, data, cancellationToken);
-            string body = await response.Content.ReadAsStringAsync();
+            HttpResponseMessage response = await MakeRequestInnerAsync(null, HttpCompletionOption.ResponseContentRead, method, path, queryString, null, data, cancellationToken).ConfigureAwait(false);
+
+            string body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
             HandleIfErrorResponse(response.StatusCode, body, errorHandlers);
+
             return new DockerApiResponse(response.StatusCode, body);
         }
 
         internal async Task<Stream> MakeRequestForStreamAsync(IEnumerable<ApiResponseErrorHandlingDelegate> errorHandlers, HttpMethod method, string path, IQueryString queryString, IDictionary<string, string> headers, IRequestContent data, CancellationToken cancellationToken)
         {
-            HttpResponseMessage response = await MakeRequestInnerAsync(TimeSpan.FromMilliseconds(Timeout.Infinite), HttpCompletionOption.ResponseHeadersRead, method, path, queryString, headers, data, cancellationToken);
+            HttpResponseMessage response = await MakeRequestInnerAsync(InfiniteTimeout, HttpCompletionOption.ResponseHeadersRead, method, path, queryString, headers, data, cancellationToken).ConfigureAwait(false);
+
             HandleIfErrorResponse(response.StatusCode, null, errorHandlers);
-            return await response.Content.ReadAsStreamAsync();
+
+            return await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
         }
 
-        private async Task<HttpResponseMessage> MakeRequestInnerAsync(TimeSpan? requestTimeout, HttpCompletionOption completionOption, HttpMethod method, string path, IQueryString queryString, IDictionary<string, string> headers, IRequestContent data, CancellationToken cancellationToken)
+        private Task<HttpResponseMessage> MakeRequestInnerAsync(TimeSpan? requestTimeout, HttpCompletionOption completionOption, HttpMethod method, string path, IQueryString queryString, IDictionary<string, string> headers, IRequestContent data, CancellationToken cancellationToken)
         {
-            using (HttpClient client = this.GetHttpClient())
+            HttpRequestMessage request = PrepareRequest(method, path, queryString, headers, data);
+
+            if (requestTimeout.HasValue)
```

Comment:
@ItsVeryWindy ah you're right. and I bet `CreateTimeoutToken(InfiniteTimeout)` is not a good idea either. Looks good then.